sdk/python/feast/pyspark/abc.py

```diff
@@ -536,3 +547,7 @@ def get_job_by_id(self, job_id: str) -> SparkJob:
     @abc.abstractmethod
     def list_jobs(self, include_terminated: bool) -> List[SparkJob]:
         raise NotImplementedError
+
+    @abc.abstractmethod
+    def list_jobs_by_hash(self, include_terminated: bool) -> Dict[str, SparkJob]:
+        raise NotImplementedError
```

Comment:
I don't feel super strongly about this, but it may be a bit more elegant to have `StreamIngestionJob` interface implement `get_job_hash()`, and not extend `JobLauncher` interface at all. 

Otherwise we have a situation where some job attributes are part of the XXJob interface (like output location for `RetrievalJob`) and some are not (like job hash here).

Reply:
sure, I can do that

Reply:
hmm I thought about this and I think one big difference is that get_output_file_uri is a user-facing feature, while job hash is purely implementation, so I think it's better to not expose that as part of the job api. It'll also be more work to properly expose that method (e.g. we'd need to implement RemoteStreamIngestionJob too)

Reply:
I mean `StreamIngestionJob` interface is not really public anyway right now. It is just a bit unorthogonal to have two `list_jobs*` almost-identical methods in the launcher interface where we could've made that part of the job itself.

This said, I don't feel super strongly so we can leave it as is too.

Reply:
What do you mean it's not public? `client.start_stream_to_online_ingestion` returns that right?

Reply:
ah you're right